src/analysis/sum_integral_comparisons.lean

```diff
@@ -0,0 +1,169 @@
+/-
+Copyright (c) 2022 Kevin H. Wilson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin H. Wilson
+-/
+import measure_theory.measure.measure_space_def
+import measure_theory.integral.interval_integral
+import algebra.order.floor
+import data.set.function
+import analysis.special_functions.integrals
+
+/-!
+# Comparing sums and integrals
+
+## Summary
+
+It is often the case that error terms in analysis can be computed by comparing
+an infinite sum to the improper integral of an antitone function. This file will eventually enable
+that.
+
+At the moment it contains four lemmas in this direction: `antitone_on.integral_le_sum`,
+`antitone_on.sum_le_integral` and versions for monotone functions, which can all be paired
+with a `filter.tendsto` to estimate some errors.
+
+`TODO`: Add more lemmas to the API to directly address limiting issues
+
+## Main Results
+
+* `antitone_on.integral_le_sum`: The integral of an antitone function is at most the sum of its
+  values at integer steps aligning with the left-hand side of the interval
+* `antitone_on.sum_le_integral`: The sum of an antitone function along integer steps aligning with
+  the right-hand side of the interval is at most the integral of the function along that interval
+* `monotone_on.integral_le_sum`: The integral of a monotone function is at most the sum of its
+  values at integer steps aligning with the right-hand side of the interval
+* `monotone_on.sum_le_integral`: The sum of a monotone function along integer steps aligning with
+  the left-hand side of the interval is at most the integral of the function along that interval
+
+## Tags
+
+analysis, comparison, asymptotics
+-/
+
+open set measure_theory.measure_space
+open_locale big_operators
+
```

Comment:
You use the same implicit arguments a lot: you should declare them once using `variables` and remove them from the lemmas.
```suggestion

variables {x₀ : ℝ} {a b : ℕ} {f : ℝ → ℝ}

```